packages/react/src/utils/questionnaire.ts

```diff
@@ -95,6 +96,119 @@ export function isQuestionEnabled(
   return enableBehavior !== 'any';
 }
 
+export function evaluateCalculatedExpressionsInQuestionnaire(
+  items: QuestionnaireItem[],
+  response: QuestionnaireResponse | undefined
+): QuestionnaireResponseItem[] {
+  return items
+    .map((item): QuestionnaireResponseItem | null => {
+      if (item.item) {
+        return {
+          ...item,
+          item: evaluateCalculatedExpressionsInQuestionnaire(item.item, response),
+        };
+      } else {
+        const calculatedValue = evaluateCalculatedExpression(item, response);
+        if (!calculatedValue) {
+          return null;
+        }
+
+        const answer = typedValueToResponseItem(item, calculatedValue);
+
+        if (!answer) {
+          return null;
+        }
+
+        return {
+          id: item?.id,
+          linkId: item?.linkId,
+          text: item.text,
+          answer: [answer],
+        };
+      }
+    })
+    .filter((item): item is QuestionnaireResponseItem => item !== null);
+}
+
+export function typedValueToResponseItem(
+  item: QuestionnaireItem,
+  value: TypedValue
+): QuestionnaireResponseItemAnswer | undefined {
+  if (!item.type) {
+    return undefined;
+  }
+
+  switch (item.type) {
+    case QuestionnaireItemType.boolean:
+      return value.type === PropertyType.boolean ? { valueBoolean: value.value } : undefined;
+    case QuestionnaireItemType.date:
+      return value.type === PropertyType.date ? { valueDate: value.value } : undefined;
+    case QuestionnaireItemType.dateTime:
+      return value.type === PropertyType.dateTime ? { valueDateTime: value.value } : undefined;
+    case QuestionnaireItemType.time:
+      return value.type === PropertyType.time ? { valueTime: value.value } : undefined;
+    case QuestionnaireItemType.url:
+      return value.type === PropertyType.url ? { valueString: value.value } : undefined;
+    case QuestionnaireItemType.text:
+      return value.type === PropertyType.string ? { valueString: value.value } : undefined;
+    case QuestionnaireItemType.attachment:
+      return value.type === PropertyType.Attachment ? { valueAttachment: value.value } : undefined;
+    case QuestionnaireItemType.reference:
+      return value.type === PropertyType.Reference ? { valueReference: value.value } : undefined;
+    case QuestionnaireItemType.quantity:
+    case QuestionnaireItemType.decimal:
+    case QuestionnaireItemType.integer:
+    case QuestionnaireItemType.string:
+      return { valueQuantity: value.value };
```

Comment:
Interesting that TypeScript didn't catch this... 😬   It's because `TypedValue.value` is type `any` 🤦 

`valueQuantity` should only be a `Quantity`.  See: https://hl7.org/fhir/R4/questionnaireresponse.html

So it would be something like:

```ts
    case QuestionnaireItemType.quantity:
      return { valueQuantity: value.value };
    case QuestionnaireItemType.decimal:
      return { valueDecimal: value.value };
    case QuestionnaireItemType.integer:
      return { valueInteger: value.value };
    case QuestionnaireItemType.string:
      return { valueString: value.value };
```

Kinda gross and verbose.  There might be a more clever way to do this, because it should be systematic, but an exhaustive `switch` statement is also okay.